packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -35,22 +37,159 @@ import {validateOptions} from '../../../utils/validate-payload';
 import {
   CategoryFacetOptions,
   categoryFacetOptionsSchema,
+  CategoryFacetSearchOptions,
 } from './headless-category-facet-options';
 import {determineFacetId} from '../_common/facet-id-determinor';
+import {FacetValueState} from '../../../features/facets/facet-api/value';
+
+export {CategoryFacetOptions, CategoryFacetSearchOptions};
 
-export {CategoryFacetOptions};
-export type CategoryFacetProps = {
+export interface CategoryFacetProps {
   /** The options for the `CategoryFacet` controller. */
   options: CategoryFacetOptions;
-};
+}
+
 /**
  * The `CategoryFacet` headless controller offers a high-level interface for designing a facet UI controller that renders values in a hierarchical fashion.
  */
-export type CategoryFacet = ReturnType<typeof buildCategoryFacet>;
+export interface CategoryFacet extends Controller {
+  facetSearch: CategoryFacetSearch;
+
+  /**
+   * Toggles the specified facet value.
+   * @param selection The facet value to toggle.
```

Comment:
Small note on `param` tags. This should parse correctly, but to be spec-compliant, we should add a dash between the param name and description. e.g.

```
@param selection - The facet value to toggle.
```

Ref: https://api-extractor.com/pages/tsdoc/tag_param/

Reply:
Interesting, I like it, that's unlike JSDoc 😯
I'll fix it.